documentation/source/examples.rst

```diff
@@ -35,6 +35,23 @@ to start and stop generation of input data.
 A :class:`.TestFactory` is used to generate the random tests.
 
 
+Matrix Multiplier
+=================
+
+The directory :file:`cocotb/examples/matrix_multiplier` contains a module for multiplying two matrices together, implemented in both VHDL and Verilog.
+
+The module takes two matrices ``A`` and ``B`` as inputs and provides the resulting matrix ``C`` as an output. When input ``i_valid`` is high on a rising clock edge, ``C`` is calculated and output, and ``o_valid`` goes high.
+
+The testbench defines a ``MatMultMonitor`` (a sub-class of :class:`.BusMonitor`) and a test case, which is provided in two versions:
+
+* ``run_test`` - For simulators that support multi-dimensional arrays
+* ``run_test_array`` -  For simulators that concatenate multi-dimensional arrays into a single long array
```

Comment:
I would find the names ``ndim_array`` and ``1dim_array`` (also in the design etc.) more intuitive.